private/tools/java/com/github/bazelbuild/rules_jvm_external/jar/IndexJar.java

```diff
@@ -54,13 +79,28 @@ public static void main(String[] args) throws IOException {
     System.out.println(new Gson().toJson(index));
   }
 
-  public static SortedSet<String> process(Path path) throws IOException {
+  public static PerJarIndexResults index(Path path) throws IOException {
     SortedSet<String> packages = new TreeSet<>();
-    try (InputStream fis = new BufferedInputStream(Files.newInputStream(path));
-        ZipInputStream zis = new ZipInputStream(fis)) {
-      try {
-        ZipEntry entry;
-        while ((entry = zis.getNextEntry()) != null) {
+    SortedMap<String, SortedSet<String>> serviceImplementations = new TreeMap<>();
+    try {
+      try (ZipFile zipFile = new ZipFile(path.toFile())) {
```

Comment:
It's not clear why you've switched to a `ZipFile` here. You can just read to EOF of `zis` to read the contents of an entry from the `ZipInputStream`, which would save the extra try-with-resources block you have later in this.